packages/@aws-cdk/toolkit-lib/lib/actions/bootstrap/index.ts

```diff
@@ -0,0 +1,204 @@
+ */
+export interface BootstrappingParameterValues {
+  /**
+   * The name to be given to the CDK Bootstrap bucket
+   * By default, a name is generated by CloudFormation
+   *
+   * @default - No value, optional argument
+   */
+  readonly bucketName?: string;
+
+  /**
+   * The ID of an existing KMS key to be used for encrypting items in the bucket
+   * By default, the default KMS key is used
+   *
+   * @default - No value, optional argument
+   */
+  readonly kmsKeyId?: string;
+
+  /**
+   * Whether or not to create a new customer master key (CMK)
+   *
+   * Only applies to modern bootstrapping
+   * Legacy bootstrapping will never create a CMK, only use the default S3 key
+   *
+   * @default false
+   */
+  readonly createCustomerMasterKey?: boolean;
+
+  /**
+   * The list of AWS account IDs that are trusted to deploy into the environment being bootstrapped
+   *
+   * @default []
+   */
+  readonly trustedAccounts?: string[];
+
+  /**
+   * The list of AWS account IDs that are trusted to look up values in the environment being bootstrapped
+   *
+   * @default []
+   */
+  readonly trustedAccountsForLookup?: string[];
+
+  /**
+   * The list of AWS account IDs that should not be trusted by the bootstrapped environment
+   * If these accounts are already trusted, they will be removed on bootstrapping
+   *
+   * @default []
+   */
+  readonly untrustedAccounts?: string[];
+
+  /**
+   * The ARNs of the IAM managed policies that should be attached to the role performing CloudFormation deployments
+   * In most cases, this will be the AdministratorAccess policy
+   * At least one policy is required if `trustedAccounts` were passed
+   *
+   * @default []
+   */
+  readonly cloudFormationExecutionPolicies?: string[];
+
+  /**
+   * Identifier to distinguish multiple bootstrapped environments
+   * The default qualifier is an arbitrary but unique string
+   *
+   * @default - 'hnb659fds'
+   */
+  readonly qualifier?: string;
+
+  /**
+   * Whether or not to enable S3 Staging Bucket Public Access Block Configuration
+   *
+   * @default true
+   */
+  readonly publicAccessBlockConfiguration?: boolean;
+
+  /**
+   * Flag for using the default permissions boundary for bootstrapping
+   *
+   * @default - No value, optional argument
+   */
+  readonly examplePermissionsBoundary?: boolean;
+
+  /**
+   * Name for the customer's custom permissions boundary for bootstrapping
+   *
+   * @default - No value, optional argument
+   */
+  readonly customPermissionsBoundary?: string;
+}
+
+/**
+ * Parameters for the bootstrapping template with flexible configuration options
+ */
+export class BootstrappingParameters {
+  /**
+   * Use default values for all parameters
+   */
+  static default() {
+    return new BootstrappingParameters();
+  }
+
+  /**
+   * Use custom parameters and fall back to default values
+   */
+  static custom(params: BootstrappingParameterValues) {
+    return new BootstrappingParameters(params);
+  }
```

Comment:
The split between `default` (= empty) and `custom` seems unnecessary. What I mean is a way to say "usePreviousParameters` or overwrite. Essentially what `deploy` is doing with `StackParamter`.

Reply:
Changed this to mirror the deployment option StackParameters instead, since deployBootstrap is itself a deployment that uses the usePreviousParameters key.